packages/ethereum-contracts/contracts/mocks/ERC777SenderRecipientMock.sol

```diff
@@ -170,6 +170,12 @@ contract ERC777SenderRecipientMock is Context, IERC777Sender, IERC777Recipient,
 contract ERC777RecipientReverting is IERC777Recipient, IERC1820Implementer {
     bytes32 constant private _TOKENS_RECIPIENT_INTERFACE_HASH = keccak256("ERC777TokensRecipient");
 
+    // allow to use the hook for this contract itself
+    constructor() {
```

Comment:
FWIW, this contract is taken from a OZ implementation. WE are not keeping track of the upstream changes so far.